From 8a945d47fca1bca6ff229b8816c9c58343a91280 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Thu, 7 Dec 2023 10:18:55 +0100
Subject: [PATCH 0701/1141] media: verisilicon: do not limit max burst length
 of h264 encoder

H264 encoding requires max burst length to 16.

Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Change-Id: I46ac5ab11f70849a0cb0a94763ec72735e780f32
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/344690
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
---
 drivers/media/platform/verisilicon/hantro_h1.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/media/platform/verisilicon/hantro_h1.c
+++ b/drivers/media/platform/verisilicon/hantro_h1.c
@@ -312,6 +312,7 @@ void hantro_h1_set_axi_ctrl(struct hantr
 {
 	bool input_swap8, input_swap16, input_swap32;
 	u32 reg;
+	u32 max_burst_length = vpu->max_burst_length;
 
 	if (is_rgb32(&ctx->src_fmt)) {
 		input_swap32 = 0;
@@ -328,9 +329,12 @@ void hantro_h1_set_axi_ctrl(struct hantr
 		input_swap8  = 1;
 	}
 
+	if (ctx->vpu_dst_fmt->codec_mode == HANTRO_MODE_H264_ENC)
+		max_burst_length = 16;
+
 	reg =     H1_REG_AXI_CTRL_OUTPUT_SWAP16
 		| (input_swap16 ? H1_REG_AXI_CTRL_INPUT_SWAP16 : 0)
-		| H1_REG_AXI_CTRL_BURST_LEN(vpu->max_burst_length)
+		| H1_REG_AXI_CTRL_BURST_LEN(max_burst_length)
 		| H1_REG_AXI_CTRL_OUTPUT_SWAP32
 		| (input_swap32 ? H1_REG_AXI_CTRL_INPUT_SWAP32 : 0)
 		| H1_REG_AXI_CTRL_OUTPUT_SWAP8
